### PR TITLE
Handle invalid class constant

### PR DESCRIPTION
**Pull request by JSAbrahams**
Depends on #312 

Just make sure the exception is caught so it doesn't break the pipeline

### CODE REVIEW

File: modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt [hunk 1]
```diff
@@ -179,7 +182,8 @@ private class UserUsageValueFilterRule(private val libraryProject: JavaProject)
     override fun apply(value: Constant) = try {
         value !is ClassConstant || isNotUserClass(value.toSootType())
     } catch (e: RuntimeException) {
-        throw FilterException(e)
+        logger.error { "Invalid constant found: \"$value\"." }
```
Comment (FWDekker) [inline modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Euh, I know I said error, but I think warning is more appropriate. ^^'

Comment (JSAbrahams) [reply modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Oh wait yes good point.

File: modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt [hunk 2]
```diff
@@ -175,7 +178,13 @@ private class UserUsageValueFilterRule(private val libraryProject: JavaProject)
     override fun apply(value: InstanceFieldRef) = isNotUserClass(value.field.declaringClass)
     override fun apply(value: StaticFieldRef) = isNotUserClass(value.field.declaringClass)
 
-    override fun apply(value: Constant) = value !is ClassConstant || isNotUserClass(value.toSootType())
+    @Suppress("TooGenericExceptionCaught")
```
Comment (FWDekker) [inline modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Please add a comment explaining why you suppressed this warning.

Comment (FWDekker) [inline modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Please add a comment explaining why you suppressed this warning.

**Pull request closed without merging**